File: Euclid/Game/Catalogue/CatalogueItem.cs
using Euclid.Game.Entity.Humanoid;
using Euclid.Network.Streams.Util;
using Euclid.Storage.Database.Data;
using System;
using System.Collections.Generic;

namespace Euclid.Game
{
    public class CatalogueItem : IEntity
    {
        #region Properties

        public CatalogueData Data { get; }
        public virtual ItemDefinition Definition => ItemManager.Instance.GetDefinition(Data.DefinitionId);
       
        #endregion

        #region Constructors

        public CatalogueItem(CatalogueData data)
        {
            Data = data;
        }

        #endregion

    }
}

File: euclid/src/main/java/euclid/game/catalogue/CatalogueItem.java
package euclid.game.catalogue;

import euclid.game.entity.IEntity;
import euclid.game.item.ItemDefinition;
import euclid.game.item.ItemManager;
import euclid.storage.database.data.CatalogueData;

public class CatalogueItem implements IEntity {

    private final CatalogueData data;

    public CatalogueItem(CatalogueData data) {
        this.data = data;
    }

    public CatalogueData getData() {
        return data;
    }

    public ItemDefinition getDefinition() {
        return ItemManager.getInstance().getDefinition(data.getDefinitionId());
    }
}
